index.bs

```diff
@@ -419,58 +437,131 @@ When this method is invoked, the user agent MUST execute the following algorithm
         <a>DOMException</a> whose name is "SecurityError", and terminate this algorithm.
 
-4. Process each element of {{cryptoParameters}} using the following steps, to produce a new sequence |normalizedParameters|.
-    - Let |current| be the currently selected element of {{cryptoParameters}}.
-    - If `current.type` does not contain a {{ScopedCredentialType}} supported by this implementation, then stop processing
-        |current| and move on to the next element in {{cryptoParameters}}.
-    - Let |normalizedAlgorithm| be the result of <a>normalizing an algorithm</a> [[!WebCryptoAPI]],
-        with |alg| set to `current.algorithm` and |op| set to 'generateKey'. If an error occurs during this
-        procedure, then stop processing |current| and move on to the next element in {{cryptoParameters}}.
-    - Add a new object of type {{ScopedCredentialParameters}} to |normalizedParameters|, with |type| set to `current.type` and
+4. Let |normalizedParameters| be a new [=list=].
+5. [=list/For each=] |current| of {{cryptoParameters}}:
+    - If <code>|current|.{{ScopedCredentialParameters/type}}</code> does not
+        contain a {{ScopedCredentialType}} supported by this implementation,
+        then [=continue=].
+    - Let |normalizedAlgorithm| be the result of <a>normalizing an algorithm</a>
+        [[!WebCryptoAPI]], with |alg| set to
+        <code>|current|.{{algorithm}}</code> and |op| set to `"generateKey"`. If
+        an error occurs during this procedure, then [=continue=].
+    - [=list/Append=] a new object of type {{ScopedCredentialParameters}} to
+        |normalizedParameters|, with |type| set to
+        <code>|current|.{{ScopedCredentialParameters/type}}</code> and
         |algorithm| set to |normalizedAlgorithm|.
 
-5. If |normalizedAlgorithm| is empty and {{cryptoParameters}} was not empty, cancel the timer started in step 2, reject
-    |promise| with a DOMException whose name is "NotSupportedError", and terminate this algorithm.
+6. If |normalizedAlgorithm| is empty and {{cryptoParameters}} was not empty,
+    cancel the timer started in step 2, [=reject=] |promise| with a
+    {{DOMException}} whose name is "{{NotSupportedError}}", and terminate this
+    algorithm.
+
+7. Let |clientExtensions| be a new [=list=].
+8. If the {{ScopedCredentialOptions/extensions}} member of {{options}} is
+    <a>present</a>, then [=map/for each=] |extension| → |argument| of
+    <code>{{options}}.{{ScopedCredentialOptions/extensions}}</code>:
+    1. If |extension| is not supported by this client platform, [=continue=].
+    2. Let |result| be the result of running |extension|'s [=client processing=] algorithm on |argument|.
+    3. If the algorithm returned an error, [=continue=].
+    4. [=list/Append=] |result| to |clientExtensions|.
+
+9. Let |clientData| be a new {{ClientData}} instance whose fields are:
+    : {{challenge}}
+    :: The [=base64url encoding=] of {{attestationChallenge}}
+    : {{origin}}
+    :: |rpId|
```

Comment:
We'll get the fix in whatwg/html; w3c is just where the bug happened to get filed due to #257.